teslajsonpy/energy.py

```diff
@@ -203,7 +203,7 @@ async def set_operation_mode(self, real_mode: str) -> None:
             path_vars={"site_id": self.energysite_id},
             default_real_mode=real_mode,
         )
-        if data and data["response"]["code"] == 201:
+        if data and data["response"] and data["response"]["Code"] and (len(str(data["response"]["Code"])) == 3) and (str(data["response"]["Code"])[:1] == "2"):
```

Comment:
Thanks. This is quite complex and brittle if the keys change again.

1. First use dict.get(). data.get("response") will be None if the key doesn't exist. You can nest it further. 
```python
code = data.get("response", {}).get("Code", 0) or data.get("response", {}).get("code", 0) 
```
2. If you're dealing with http codes, then just use number ranges that are valid. e.g., 
```python
code >=200 and code < 300
```
, or whatever you're looking for.

Double check the syntax, I've been coding cpp all day so may have forgotten.